client/ayon_blender/plugins/load/load_blend_link.py

```diff
@@ -0,0 +1,209 @@
+        library = self._get_library_from_collection(collection)
+        if library:
+            linked_to_coll = self._has_linked_to_existing_collections(library)
+            if not linked_to_coll:
+                bpy.data.libraries.remove(library)
+        else:
+            # Ensure the collection is linked to the scene's master collection
+            scene_collection = bpy.context.scene.collection
+            for col in collection.children:
+                scene_collection.children.unlink(col)
+        # remove the container collection
+        bpy.data.collections.remove(collection)
+
+        return True
+
+    def _get_library_from_collection(
+            self, collection: bpy.types.Collection) -> bpy.types.Library:
+        """Get the library from the collection."""
+        for child in collection.children:
+            if child.library:
+                # No override library
+                return child.library
+            # With override library
+            elif child.override_library and child.override_library.reference:
+                return child.override_library.reference.library
+
+        return None
+
+    def _has_linked_to_existing_collections(
+            self, library: bpy.types.Library) -> bool:
+        """Check if any collection loaded by link
+        scene loader linked to the same library.
+        """
+        existing_collections = [
+            container["node"] for container in ls()
+            if container["loader"] == str(
+                self.__class__.__name__)
+        ]
+        match_count = list(
+            coll for coll in existing_collections
+            if self._get_library_from_collection(coll) == library
+        )
+
+        return len(match_count) > 1
+
+    def _get_or_build_library_by_path(self, libpath: str) -> bpy.types.Library:
+        """Get the library by filepath. If there is no any
+        associated library to the path, the related library is loaded
+        accordingly."""
+        lib_name = os.path.basename(libpath)
+        for library in bpy.data.libraries:
+            if lib_name == library.name_full:
+                return library
+
+        with bpy.data.libraries.load(libpath, link=True, relative=False) as (
+            data_from,
+            data_to
+        ):
+            if data_from.collections:
+                data_to.collections = data_from.collections
+            elif data_from.objects:
+                data_to.objects = data_from.objects
+
+        # Return the newly loaded library
+        for library in bpy.data.libraries:
+            if lib_name == library.name_full:
+                return library
+
+    def _update_library_path_by_node(self, collection:bpy.types.Collection, libpath:str):
+        for node in collection.children:
+            if node and node.library:
+                node.library.name = os.path.basename(libpath)
+                node.library.filepath = libpath
+                node.library.reload()
```

Comment:
Why would `node` not be valid? :P Is the `if node`.

Also, wouldn't ALL the nodes of the loaded collection essentially be using the same library object? If so, you're now reloading the library TONS of times, which will likely be slow?

Should we instead be doing a first loop where we get all the libraries (in a unique manner so we don't have duplicates). And then just do:
```python
for library in libraries:
    library.name = os.path.basename(libpath)
    library.filepath = libpath
    library.reload()
```

Reply:
It works to load multiple libraries but for some reasons it doesn't update things we want to. But we can try to test with the commit again.

Reply:
If certain code is convoluted or odd with the reason because "the normal way is buggy" please add a comment to the code describing why the odd workaround is needed.